src/main/java/uk/gov/companieshouse/itemhandler/service/FilingHistoryDescriptionProviderService.java

```diff
@@ -81,7 +81,9 @@ public String mapFilingHistoryDescription(String descriptionKey, Map<String, Obj
             return null;
         }
         String description = getFilingHistoryDescriptionWithKey(descriptionKey);
-        if(filingHistoryDescriptionValues.containsKey("description")) {
+        if(filingHistoryDescriptionValues == null){
```

Comment:
The unit tests help understand the intended behaviour of the function. 👍  It might help to explain the overall logic and the special cases in a comment or the function JavaDoc too.

Reply:
Yeh that's a good point, it's become fairly complex now. I'll update the PR.